borgbase_ssh.py

```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/python
+
+from pprint import pprint
+
+ANSIBLE_METADATA = {
+		'metadata_version': '1.1',
+		'status': ['preview'],
+		'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: borgbase_ssh
+
+short_description: Module for managing SSH keys in borgbase.
+
+version_added: "2.4"
+
+description:
+		- "Module for managing SSH keys in borgbase."
+
+options:
+		email:
+				description:
+						- The email address associated with the borgbase account.
+				required: true
+		password:
+				description:
+						- The password for the borgbase account.
+				required: true
+		state:
+				description:
+						- 'present' to ensure the key exists, 'absent' to ensure it doesn't. When removing key
+						match is carried out based on key name only. When adding key, if a key exists with the
+						same name but different content, the key will be silently replaced.
+				default: present
+				choices: [ absent, present ]
+		name:
+				description:
+						- The SSH key name.
+				required: true
+		key:
+				description:
+						- The SSH public key (required if state is 'present').
+				required: false
+
+author:
+		- Andy Hawkins (@adhawkinsgh)
+'''
+
+EXAMPLES = '''
+- name: Read key from file
+  slurp:
+    src: ~/.ssh/id_rsa.pub
+  register: ssh_key
+  check_mode: yes
+
+- name: Create key
+  borgbase_ssh:
+    state: present
+    email: "{{ borgbase_email }}"
+    password: "{{ borgbase_password }}"
+    name: "{{ whoami.stdout }}@{{ ansible_hostname }}"
+    key: "{{ ssh_key['content'] | b64decode }}"
+  register: borgbase_key
+
+- name: Dump create results
+  debug:
+    var: borgbase_key.key_id
+'''
+
+RETURN = '''
+key_id:
+		description: The ID of the key that was created or deleted
+		type: int
+		returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+REQUESTS_IMP_ERR = None
+try:
+	import requests
+	HAS_REQUESTS = True
+except:
+	import traceback
+
+	HAS_REQUESTS = False
+	REQUESTS_IMP_ERR = traceback.format_exc()
+
+LOGIN = '''
+mutation login(
+	$email: String!
+	$password: String!
+	$otp: String
+	) {
+		login(
+			username: $email
+			password: $password
+			otp: $otp
+		) {
+			user {
+				id
+			}
+		}
+}
+'''
+
+SSH_LIST = '''
+query data {
+	sshList {
+		id
+		name
+		keyData
+	}
+}
+'''
+
+SSH_ADD = '''
+mutation sshAdd(
+	$name: String!
+	$keyData: String!
+	) {
+		sshAdd(
+			name: $name
+			keyData: $keyData
+		) {
+			keyAdded {
+				id
+				name
+				hashMd5
+				keyType
+				bits
+			}
+		}
+}
+'''
+
+SSH_DELETE = '''
+mutation sshDelete($id: Int!) {
+  sshDelete(id: $id) {
+    ok
+  }
+}
+'''
+
+class GraphQLClient:
```

Comment:
Don't duplicate code, move common code in `module_utils` (if `module_utils/urls.py` isn't used, [`borgbase_api_client.client`](https://github.com/borgbase/borgbase-api-client/blob/e90dc5557ef3a1a372d578c1f8b324d48c74cd02/borgbase_api_client/client.py#L5) should be imported).

Reply:
I'm not sure what you're saying here. I thought that ansible modules had to be standalone, single files so that they can be easily 'deployed' to the remote server? Are you saying I can create a 'common' file that will be used by both modules?

Reply:
> Are you saying I can create a 'common' file that will be used by both modules?

Yes, see: https://docs.ansible.com/ansible/latest/dev_guide/overview_architecture.html?highlight=module_utils#module-utilities

Reply:
Perfect, thanks. Will look in to that.

Reply:
Addressed in #8